cognite/powerops/resync/models/_shared_v1_v2/production_model.py

```diff
@@ -86,31 +84,70 @@ def _plant_to_inlet_reservoir_breadth_first_search(
     Optional[str]
         The name of the reservoir connected to the plant, or None if no reservoir was found.
     """
+
+    def get_connection_path_from_last_visited(visited_paths: list, last_visited_id: int) -> list:
+        for connection in visited_paths:
+            if connection[-1] == last_visited_id:
+                connection_path_index = visited_paths.index(connection)
+                return visited_paths[connection_path_index]
+
+    def calculate_losses_from_conncetion_path(all_junctions, all_tunnels, connection_by_id, connection_path):
+        sum_losses = 0
+        order_to_loss_factor_key = {0: "loss_factor_1", 1: "loss_factor_2"}
+        for connection_id in connection_path:
+            connection = connection_by_id[connection_id]
+            if connection.get("to_type") == "junction":
+                if connection.get("order") in order_to_loss_factor_key:
+                    junction_name = connection["to"]
+                    junction_losses = all_junctions[junction_name]
+                    loss_order = connection["order"]
+                    sum_losses += junction_losses[order_to_loss_factor_key[loss_order]]
+            elif connection.get("to_type") == "tunnel":
+                tunnel_name = connection["to"]
+                tunnel_loss = all_tunnels[tunnel_name]["loss_factor"]
+                sum_losses += tunnel_loss
```

Comment:
Does this function only work if we have the "to_type" concept? I know that is not the case for some SHOP model files